ignore/src/walk.rs

```diff
@@ -1253,6 +1278,34 @@ fn check_symlink_loop(
     Ok(())
 }
 
+fn skip_filesize(
+    max_filesize: Option<u64>,
+    path: &Path,
+    ent: &Option<Metadata>,
+    is_dir: bool
+) -> bool {
```

Comment:
This function cannot accept a `Metadata` because that implies *every* file is stat'd regardless of whether a max filesize option is set or not. You need to move the metadata access to occur only after you know there is a file size filter.